mmv1/products/integrations/AuthConfig.yaml

```diff
@@ -0,0 +1,470 @@
+      - !ruby/object:Api::Type::NestedObject
+        name: 'oauth2AuthorizationCode'
+        description: |
+          OAuth2 authorization code credential.
+        conflicts:
+          - decryptedCredential.0.username_and_password
+          - decryptedCredential.0.oauth2_client_credentials
+          - decryptedCredential.0.jwt
+          - decryptedCredential.0.auth_token
+          - decryptedCredential.0.service_account_credentials
+          - decryptedCredential.0.oidc_token
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'clientId'
+            description: |
+              The client's id.
+          - !ruby/object:Api::Type::String
+            name: 'clientSecret'
+            description: |
+              The client's secret.
+          - !ruby/object:Api::Type::String
+            name: 'scope'
+            description: |
+              A space-delimited list of requested scope permissions.
+          - !ruby/object:Api::Type::String
+            name: 'authEndpoint'
+            description: |
+              The auth url endpoint to send the auth code request to.
+          - !ruby/object:Api::Type::String
+            name: 'tokenEndpoint'
+            description: |
+              The token url endpoint to send the token request to.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'oauth2ClientCredentials'
+        description: |
+          OAuth2 client credentials.
+        conflicts:
+          - decryptedCredential.0.username_and_password
+          - decryptedCredential.0.oauth2_authorization_code
+          - decryptedCredential.0.jwt
+          - decryptedCredential.0.auth_token
+          - decryptedCredential.0.service_account_credentials
+          - decryptedCredential.0.oidc_token
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'clientId'
+            description: |
+              The client's ID.
+          - !ruby/object:Api::Type::String
+            name: 'clientSecret'
+            description: |
+              The client's secret.
+          - !ruby/object:Api::Type::String
+            name: 'tokenEndpoint'
+            description: |
+              The token endpoint is used by the client to obtain an access token by presenting its authorization grant or refresh token.
+          - !ruby/object:Api::Type::String
+            name: 'scope'
+            description: |
+              A space-delimited list of requested scope permissions.
+          - !ruby/object:Api::Type::NestedObject
+            name: 'tokenParams'
+            description: |
+              Token parameters for the auth request.
+            properties:  # ParameterMap
+              - !ruby/object:Api::Type::Array
+                name: 'entries'
+                description: |
+                  A list of parameter map entries.
+                item_type: !ruby/object:Api::Type::NestedObject
+                  properties:  # ParameterMapEntry
```

Comment:
I noticed some field omissions here- can you elaborate on why we only support a subset of the configurable values? (It's likely not an issue, just want to make sure I'm not missing anything)